ovn_k8s/modes/overlay.py

```diff
@@ -438,3 +467,332 @@ def sync_services(self, services):
+        # an external id for the ACL as well (acl-add won't return the ACL id)
+        kwargs['pod_id'] = pod_id
+        external_ids = " ".join('external_ids:%s=%s' % (k, v)
+                                for k, v in kwargs.items())
+        command = ('-- --id=@acl_id create ACL action=%s direction=to-lport '
+                   'priority=%d match="%s" external_ids:lport_name=%s '
+                   '%s -- add Logical_Switch %s acls @acl_id' %
+                   (action, priority, match, lport_name,
+                    external_ids, ls_name))
+        command_items = tuple(shlex.split(command))
+        ovn_nbctl(*command_items)
+
+    def whitelist_pod_traffic(self, pod_id, namespace, pod_name):
+        lport_name = self._build_logical_port_name(namespace, pod_name)
+        lport_uuid = ovn_nbctl("--data=bare", "--no-heading",
+                               "--columns=_uuid", "find",
+                               "logical_switch_port",
+                               "name=%s" % lport_name)
+        lswitch_name = ovn_nbctl("--data=bare", "--no-heading",
+                                 "--columns=_uuid", "find",
+                                 "logical_switch",
+                                 "ports{>=}%s" % lport_uuid)
+        self._create_acl(pod_id, lswitch_name, lport_name,
+                         DEFAULT_ACL_PRIORITY,
+                         r'outport\=\=\"%s\"\ &&\ ip' % lport_name,
+                         'allow-related')
+        vlog.dbg("Pod %s (Namespace:%s): Traffic whitelisted"
+                 % (pod_name, namespace))
+
+    def _create_acl_match(self, lport_name, pseudo_acl):
+        # TODO(salv-orlando): move this function to the util module
+        src_match = None
+        ports_match = None
+        # NOTE: We do not validate that the protocol names are valid
+        ports_clause = pseudo_acl[1] or []
+        from_clause = pseudo_acl[2]
+        if from_clause and from_clause != '*':
+            # '*' means every address matches and therefore no source IP
+            # match should be added to the ACL
+            src_match = r"ip4.src\=\=\{%s\}" % from_clause
+        match_items = set()
+        protocol_port_map = {}
+        for (protocol, port) in ports_clause:
+            ports = protocol_port_map.setdefault(protocol, set())
+            if port:
+                ports.add(port)
+        for protocol, ports in protocol_port_map.items():
+            if ports:
+                item_match_str = (r"%s.dst\=\=\{%s\}" %
+                                  (protocol.lower(), ",".join(
+                                      [str(port) for port in ports
+                                       if port is not None])))
+            else:
+                item_match_str = protocol.lower()
+            match_items.add(item_match_str)
+        ports_match = "\ ||\ ".join(match_items)
+        if ports_match and src_match:
+            policy_match = "%s\ &&\ %s" % (ports_match, src_match)
```

Comment:
Looking back at the yml policy files I used to test this code path went untested. You are therefore right I think